mne/beamformer/_lcmv.py

```diff
@@ -102,52 +133,86 @@ def _apply_lcmv(data, info, tmin, forward, noise_cov, data_cov, reg,
 
-            Wk[:] = np.dot(max_ori, Wk)
-            Ck = np.dot(max_ori, np.dot(Ck, max_ori))
             is_free_ori = False
 
-        if is_free_ori:
-            # Free source orientation
-            Wk[:] = np.dot(linalg.pinv(Ck, 0.1), Wk)
+        # NAI without orientation selection
+        elif weight_norm == 'nai' and pick_ori != 'max-power':
+            raise ValueError('Weight normalization with neural activity index '
+                             'is not implemented yet with free or fixed '
+                             'orientation.')
+
+        # all other combinations are handled here:
         else:
-            # Fixed source orientation
-            Wk /= Ck
+            if pick_ori == 'max-power':
+                eig_vals, eig_vecs = linalg.eigh(Ck)
+
+                # Choosing the eigenvector associated with the middle
+                # eigenvalue. The middle and not the minimal eigenvalue is used
+                # because MEG is insensitive to one (radial) of the three
+                # dipole orientations and therefore the smallest eigenvalue
+                # reflects mostly noise.
+                for ii in range(3):
+                    if ii != eig_vals.argmax() and ii != eig_vals.argmin():
+                        idx_middle = ii
+
+                # TODO: The eigenvector associated with the smallest eigenvalue
+                # should probably be used when using combined EEG and MEG data
+                max_ori = eig_vecs[:, idx_middle]
+
+                Wk[:] = np.dot(max_ori, Wk)
+                Ck = np.dot(max_ori, np.dot(Ck, max_ori))
+                is_free_ori = False
+
+            if is_free_ori:
+                # Free source orientation
+                Wk[:] = np.dot(linalg.pinv(Ck, 0.1), Wk)
+            else:
+                # Fixed source orientation
+                Wk /= Ck
 
     # Pick source orientation maximizing output source power
     if pick_ori == 'max-power':
         W = W[0::3]
+    else:
+        if weight_norm is not None:
+            # Preparing noise normalization
+            noise_norm = np.sum(W ** 2, axis=1)
+            if is_free_ori:
+                noise_norm = np.sum(noise_norm.reshape(-1, 3), axis=1)
+            noise_norm = np.sqrt(noise_norm)
+
+        # Pick source orientation normal to cortical surface
+        if pick_ori == 'normal':
+            W = W[2::3]
+            is_free_ori = False
 
-    # Preparing noise normalization
-    noise_norm = np.sum(W ** 2, axis=1)
-    if is_free_ori:
-        noise_norm = np.sum(np.reshape(noise_norm, (-1, 3)), axis=1)
-    noise_norm = np.sqrt(noise_norm)
-
-    # Pick source orientation normal to cortical surface
-    if pick_ori == 'normal':
-        W = W[2::3]
-        is_free_ori = False
-
-    # Applying noise normalization
-    noise_norm_inv = 1. / noise_norm
-    noise_norm_inv[noise_norm == 0.] = 0.
-    if not is_free_ori:
-        W *= noise_norm_inv[:, None]
+        if weight_norm is not None:
```

Comment:
Mh, I think that clashes with the way it's done. for the  normalization as implemented here the filter needs to be already calculated, while we do the normalization in the case of max-power while calculating the filter.